wkcuber/api/Dataset.py

```diff
@@ -72,40 +101,58 @@ def get_layer(self, layer_name) -> Layer:
             )
         return self.layers[layer_name]
 
-    def add_layer(self, layer_name, category, dtype=None, num_channels=None, **kwargs):
-        if dtype is None:
-            dtype = np.dtype("uint8")
+    def add_layer(
+        self, layer_name, category, dtype_per_layer=None, num_channels=None, **kwargs
+    ):
         if num_channels is None:
             num_channels = 1
-
-        # normalize the value of dtype in case the parameter was passed as a string
-        dtype = np.dtype(dtype)
+        if dtype_per_layer is None:
+            dtype_per_layer = "uint" + str(8 * num_channels)
```

Comment:
Maybe extract the 8 into a `DEFAULT_BIT_DEPTH` constant at the top of the module.